app/helpers/patient_query_helper.rb

```diff
@@ -281,18 +281,52 @@ def advanced_filter(patients, filters, tz_offset)
         patients = advanced_filter_relative_date(patients, :symptom_onset, filter, tz_diff, :date)
       when 'continous-exposure'
         patients = patients.where(continuous_exposure: filter[:value].present? ? true : [nil, false])
+      when 'close-contact-with-known-case'
+        case filter[:additionalFilterOption]
+        when 'Exact Match'
+          patients = if filter[:value].blank?
+                       patients.where(contact_of_known_case_id: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id = ?', filter[:value])
+                     end
+        when 'Contains'
+          patients = if filter[:value].blank?
+                       patients.where(contact_of_known_case_id: [nil, ''])
+                     else
+                       patients.where('patients.contact_of_known_case_id like ?', "%#{filter[:value]}%")
+                     end
+        when 'Multiple'
+          if filter[:value].blank?
+            patients = patients.where(contact_of_known_case_id: [nil, ''])
+          else
+            value_array = filter[:value].split(/\s*,\s*/)
+            filtered_patient_ids = []
+            value_array.each do |value|
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id = ?', value).pluck(:id)
+              filtered_patient_ids += patients.where('patients.contact_of_known_case_id like ?', "#{value},%").pluck(:id)
```

Comment:
I was under the impression the "Multiple" option was checking exact matches between multiple items but instead it's exact matches OR contains?